src/treeview/writer.js

```diff
@@ -0,0 +1,623 @@
+import CKEditorError from '../ckeditorerror.js';
+
+/**
+ * Tree model Writer class.
+ * Writer defines a high-level API for TreeView manipulations.
+ *
+ * @memberOf core.treeView
+ */
+ export default class Writer {
+	constructor() {
+		/**
+		 * Priorities map. Maps node to priority.
+		 * Nodes with priorities are considered as attributes.
+		 *
+		 * @member {WeakMap} core.treeView.Writer#_priorities
+         * @protected
+         */
+		this._priorities = new WeakMap();
+	}
+
+	/**
+	 * Returns `true` if provided node is a `container` node.
+	 * Node is considered as `container` when it is a {@link core.treeView.Element Element} instance and has no priority
+	 * defined.
+	 *
+	 * @param {core.treeView.Element} node Node to check.
+	 * @returns {Boolean} `True` if provided node is a container.
+     */
+	isContainer( node ) {
+		const isElement = node instanceof Element;
+
+		return isElement && !this._priorities.has( node );
+	}
+
+	/**
+	 * Returns `true` if provided node is an `attribute` node.
+	 * Node is considered as `attribute` when it is a {@link core.treeView.Element Element} instance and has priority
+	 * defined.
+	 *
+	 * @param {core.treeView.Element} node Node to check.
+	 * @returns {Boolean} `True` if provided node is an attribute.
+	 */
+	isAttribute( node ) {
+		const isElement = node instanceof Element;
+
+		return isElement && this._priorities.has( node );
+	}
+
+	/**
+	 * Sets node priority. When priority is defined, node is considered as `attribute`
+	 *
+	 * @param {core.treeView.Node} node
+	 * @param {Number} priority
+     */
+	setPriority( node, priority ) {
+		this._priorities.set( node, priority );
+	}
+
+	/**
+	 * Returns node's priority.
+	 *
+	 * @param {core.treeView.Node} node Node to check its priority.
+	 * @returns {Number|undefined} Priority or `undefined` if there is no priority defined.
+     */
+	getPriority( node ) {
+		return this._priorities.get( node );
+	}
+
+	/**
+	 * Returns first parent container of specified {@link core.treeView.Position Position}.
+	 * Position's parent node is checked as first, then next parents are checked.
+	 *
+	 * @param {core.treeView.Position} position Position used as a start point to locate parent container.
+	 * @returns {core.treeView.Element|null} Parent container element or `null` if container is not found.
+	 */
+	getParentContainer( position ) {
+		let parent = position.parent;
+
+		while ( !this.isContainer( parent ) ) {
+			if ( !parent ) {
+				return null;
+			}
+			parent = parent.parent;
+		}
+
+		return parent;
+	}
+
+	/**
+	 * Breaks attribute nodes at provided position. It breaks `attribute` nodes inside `container` node.
+	 *
+	 *		<p>{foo}<b><u>{bar}|</u></b></p> -> <p>{foo}<b><u>{bar}</u></b>|</p>
+	 *		<p>{foo}<b><u>|{bar}</u></b></p> -> <p>{foo}|<b><u>{bar}</u></b></p>
+	 *		<p>{foo}<b><u>{b|ar}</u></b></p> -> <p>{foo}<b><u>{b}</u></b>|<b><u>{ar}</u></b></p>
+	 *
+	 * @see {@link core.treeView.Writer#isContainer}
```

Comment:
You do not need @link in @see tag. Change to `@see core.treeView.Writer#isContainer`.